### ISSUE TITLE
Check port 8090 is free before starting Gatsby

### ISSUE THREAD

**Issue by lavelle** [thumbtack/thumbprint#381]
Currently, if this port is already taken (usually by a previous ghost gatsby process that wasn't killed properly) then it will spend 2 minutes building the docs, and only then show an error when it tries to start the server.

Would be nice to quickly check this upfront, and show an error recommending that the user kill their ghost server.

**Comment by danoc** [comment]
Gatsby should check if the port is open and then offer to do it on a different port if not.

Maybe this isn't happening because we're forcing port 8090? We can remove that part and leave it up to Gatsby.

**Comment by lavelle** [comment]
I would rather have it locked to 8090 because then you can refresh the same browser tab after the server crashes and restarts. It's annoying to have to read a different port number every time it starts

**Comment by lavelle** [comment]
So I think just checking for the port open status before would be better

**Comment by danoc** [comment]
Going to close this one as a "won't fix." I'd rather file an issue with Gatsby if we want to improve this experience.
